Source/santa-driver/SantaDriverClient.cc

```diff
@@ -248,6 +248,28 @@ IOReturn SantaDriverClient::cache_bucket_count(
   return kIOReturnSuccess;
 }
 
+IOReturn SantaDriverClient::add_filemod_prefix_filter(
+    OSObject *target, void *reference, IOExternalMethodArguments *arguments) {
+  SantaDriverClient *me = OSDynamicCast(SantaDriverClient, target);
+  if (!me) return kIOReturnBadArgument;
+
+  const char *prefix = reinterpret_cast<const char *>(arguments->structureInput);
+
+  IOReturn ret = kIOReturnSuccess;
+  uint32_t node_count = 0;
+
+  // If the client sends a bad address the kernel will kill the client before it reaches here.
+  if (prefix[0] == '\0') {
```

Comment:
I don't like the empty string being used as a magic value to force a reset - I think the empty string should be ignored (and probably logged) and another client method should be added to do a reset.